ehrql/query_model/table_schema.py

```diff
@@ -22,15 +23,24 @@ def __post_init__(self):
         def description(self):
             return f"Possible values: {', '.join(f'`{v}`' for v in self.values)}"
 
+        def validate(self, value):
+            return self.values == value if value else False
+
     class NotNull(BaseConstraint):
         description = "Never `NULL`"
 
+        def validate(self, value):
+            return value is not None
+
     class Unique(BaseConstraint):
         description = "Always unique"
```

Comment:
This class could have a `validate` method that returns `True`, with a comment explaining that we can't actually validate a single value.